pkg/sink/sink_test.go

```diff
@@ -179,687 +179,758 @@ func checkSinkResponse(t *testing.T, resp *http.Response, elName string) {
+			sink, dynamicClient := getSinkAssets(t, tc.resources, elName, tc.webhookInterceptor)
+			ts := httptest.NewServer(http.HandlerFunc(sink.HandleEvent))
+			defer ts.Close()
+			req, err := http.NewRequest("POST", ts.URL, bytes.NewReader(tc.eventBody))
+			if err != nil {
+				t.Fatalf("error creating request: %s", err)
+			}
+			if tc.headers != nil {
+				req.Header = http.Header(tc.headers)
+			}
+			req.Header.Add("Content-Type", "application/json")
+			resp, err := http.DefaultClient.Do(req)
+			if err != nil {
+				t.Fatalf("error sending request: %s", err)
+			}
+			checkSinkResponse(t, resp, elName)
+			// Check right resources were created.
+			got := toTaskRun(t, dynamicClient.Actions())
+			// Sort actions (we do not know what order they executed in)
+			if diff := cmp.Diff(tc.want, got, cmpopts.SortSlices(func(x, y pipelinev1.TaskRun) bool {
+				return x.Name < y.Name
+			})); diff != "" {
+				t.Errorf("Created resources mismatch (-want + got): %s", diff)
+			}
+		})
 	}
 }
 
-func TestHandleEventPassesURLThrough(t *testing.T) {
-	eventBody := json.RawMessage(`{"head_commit": {"id": "testrevision"}, "repository": {"url": "testurl"}}`)
-	pipelineResource := pipelinev1alpha1.PipelineResource{
-		TypeMeta: metav1.TypeMeta{
-			APIVersion: "tekton.dev/v1alpha1",
-			Kind:       "PipelineResource",
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "my-pipelineresource",
-			Namespace: namespace,
-		},
-		Spec: pipelinev1alpha1.PipelineResourceSpec{
-			Type: pipelinev1alpha1.PipelineResourceTypeGit,
-			Params: []pipelinev1alpha1.ResourceParam{{
-				Name:  "url",
-				Value: "$(tt.params.url)",
-			}},
-		},
-	}
-	pipelineResourceBytes, err := json.Marshal(pipelineResource)
-	if err != nil {
-		t.Fatalf("Error unmarshalling pipelineResource: %s", err)
-	}
-
-	tt := bldr.TriggerTemplate("tt", namespace,
-		bldr.TriggerTemplateSpec(
-			bldr.TriggerTemplateParam("url", "", ""),
-			bldr.TriggerResourceTemplate(runtime.RawExtension{Raw: pipelineResourceBytes}),
-		))
-	tb := bldr.TriggerBinding("tb", namespace,
-		bldr.TriggerBindingSpec(
-			bldr.TriggerBindingParam("url", "$(body.repository.url)"),
-		))
-
-	el := &triggersv1.EventListener{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "el",
-			Namespace: namespace,
-		},
-		Spec: triggersv1.EventListenerSpec{
-			Triggers: []triggersv1.EventListenerTrigger{{
-				Bindings: []*triggersv1.EventListenerBinding{{Ref: "tb", Kind: "TriggerBinding"}},
-				Template: &triggersv1.EventListenerTemplate{Name: "tt"},
-				Interceptors: []*triggersv1.EventInterceptor{{
-					CEL: &triggersv1.CELInterceptor{
-						Overlays: []triggersv1.CELOverlay{
-							{Key: "new", Expression: "body.repository.url"},
-						},
-					},
-				}},
-			}},
-		},
-	}
-	secret := &corev1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "secret",
-			Namespace: namespace,
-		},
-		Data: map[string][]byte{
-			"secretKey": []byte("secret"),
-		},
-	}
+// Setup for TestHandleEvent_AuthOverride
```

Comment:
Why are these declared in the middle of the file as compared to either at the top or in inside the test func?

Reply:
We can't put it inside the TestHandleEvent_AuthOverride since these variables are used in the OverrideAuthentication method that we implement for the `fakeAuth` type below. We could put these at the top of the file too but given that these are only used for the test below, we declare them close to where they are being used.